features.md

```diff
@@ -160,3 +160,78 @@ Known limitations
 > editors, e.g. VSCode, provide a way to preview the rename changes before
 > applying them; while some just apply the changes directly.
 {:.tip}
+
+## Semantic Highlighting
+
+clangd computes semantic highlight tokens to enhance the source code theming. {.v9}
+
+![screenshot: highlight](screenshots/semantic_highlight.png)
+
+Tokens may be associated with one highlight kind and multiple highlight modifiers, as defined by the following section.
+
+### Kinds
+
+clangd supports following standard LSP token kinds:
+
+| Kind                   | LSP Name      | Note                       
+|------------------------|---------------|--------------------------
+| Variable{.v9}          | variable      |                          
+| LocalVariable{.v9}     | variable      |                          
+| StaticField{.v9}       | variable      |                          
+| Parameter{.v9}         | parameter     |                          
+| StaticMethod{.v9}      | function      |                          
+| Function{.v9}          | function      |                          
+| Method{.v9}            | method        |                          
+| Field{.v9}             | property      |                          
+| Class{.v9}             | class         |                          
+| Interface{.v12}        | interface     | Objective-C only         
+| Enum{.v9}              | enum          |                          
+| EnumConstant{.v9}      | enumMember    |                          
+| Type{.v12}             | type          |                          
+| Typedef{.v9}           | type          |                          
+| Primitive{.v9}         | type          |                          
+| TemplateParameter{.v9} | typeParameter |                          
+| Namespace{.v9}         | namespace     |                          
+| Macro{.v9}             | macro         |                          
+| Modifier{.v16}         | modifier      | `override`/`final`       
+| Operator{.v15}         | operator      | Including `new`/`delete` 
+| Bracket{.v15}          | bracket       | All `()[]{}`             
+| Label{.v15}            | label         |                          
+
+On top of that, as an extension, clangd supports following additional token kinds:
+
+| Kind               | LSP Name     | Note                           
+|--------------------|--------------|--------------------------------
+| Unknown{.v12}      | unknown      | Dependent names                
+| Concept{.v12}      | concept      |                                
+| InactiveCode{.v11} | inactiveCode | Code removed from preprocessing
```

Comment:
nit: rephrase "Code removed from preprocessing" to "Inactive preprocessor branches"